pkg/agent/core/ngt/service/ngt.go

```diff
@@ -644,6 +641,18 @@ func (n *ngt) SaveIndex(ctx context.Context) (err error) {
 }
 
 func (n *ngt) saveIndex(ctx context.Context) (err error) {
+	noice := atomic.LoadUint64(&n.nocie)
+	if atomic.LoadUint64(&n.lastNoice) == noice {
+		return
+	}
+	atomic.SwapUint64(&n.lastNoice, noice)
+	// wait for not indexing & not saving
+	for n.IsIndexing() || n.IsSaving() {
+	}
```

Comment:
i think a busy loop is not a good pattern. how about to add a sleep or use channel to wait for it?

Reply:
Great! thanks for revising.